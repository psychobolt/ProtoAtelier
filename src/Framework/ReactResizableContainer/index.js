export { default } from './Resizable.container';
