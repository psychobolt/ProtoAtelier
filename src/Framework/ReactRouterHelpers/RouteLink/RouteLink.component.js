// @flow
import React from 'react';
import { Route } from 'react-router-dom';

type Path = {
  to: string,
  default: ?boolean
};

type Props = {
  to: string | Array<Path>,
  children: any,
  selectedProp: string
};

export default ({ to, selectedProp = 'active', children }: Props) => {
  let route;
  let path;
  if (typeof to === 'string') {
    route = to;
    path = to;
  } else if (to && to.length) {
    route = (to.find(p => p.default) || to[0]).to;
    path = `(${to.map(p => p.to).reduce((prev, curr) => `${prev}|${curr}`)})`;
  } else {
    path = '';
    route = '';
  }
  return (
    <Route
      path={path}
      children={({ match, history }) => // eslint-disable-line react/no-children-prop
        React.Children.map(children, child => {
          const { onClick, ...rest } = child.props;
          return React.cloneElement(child, {
            [selectedProp]: match ? true : null,
            onClick: (event: SyntheticEvent<*>) =>
              ((onClick && onClick(event)) || true) && (!match && history.push(route)),
            ...rest,
          });
        })}
    />
  );
};
