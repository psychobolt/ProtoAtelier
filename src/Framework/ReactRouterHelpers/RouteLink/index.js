export { default } from './RouteLink.component';
