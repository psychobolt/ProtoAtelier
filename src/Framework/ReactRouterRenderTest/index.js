// @flow
import React from 'react';
import {
  Route,
  MemoryRouter,
  type Location,
  type LocationShape,
  type Match,
  type RouterHistory,
} from 'react-router-dom';
import { mount } from 'enzyme';

// a way to render any part of your app inside a MemoryRouter
// you pass it a list of steps to execute when the location
// changes, it will call back to you with stuff like
// `match` and `location`, and `history` so you can control
// the flow and make assertions.
type Callback = (props: {
  match: Match,
  location: Location,
  history: RouterHistory,
}) => {};

type Options = {
  initialEntries: Array<LocationShape | string>,
  initialIndex: number,
  steps: Array<Callback>
};

type Props = {
  children: any,
  steps: Array<Callback>,
  match: Match,
  location: Location,
  history: RouterHistory,
};

class Assert extends React.Component<Props> {
  componentDidMount() {
    this.assert();
  }

  componentDidUpdate() {
    this.assert();
  }

  assert() {
    const { steps } = this.props;
    if (steps && steps.length) {
      const nextStep = steps.shift();
      if (nextStep) {
        const { match, location, history } = this.props;
        nextStep({ match, location, history });
      }
    }
  }

  props: Props;

  render() {
    return this.props.children;
  }
}

const renderTest = (subject: any, {
  initialEntries,
  initialIndex,
  steps,
}: Options) => mount(
  <MemoryRouter
    initialIndex={initialIndex}
    initialEntries={initialEntries}
  >
    <Route render={(props) => (
      <Assert steps={steps} {...props}>
        {subject}
      </Assert>
    )}
    />
  </MemoryRouter>,
);

export default renderTest;
