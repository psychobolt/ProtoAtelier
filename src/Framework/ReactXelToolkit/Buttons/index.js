export { default as XButton } from './XButton.component';
