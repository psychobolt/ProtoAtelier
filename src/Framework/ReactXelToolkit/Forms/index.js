export { default as XInput, XInputEvent } from './XInput';
export { default as XNumberInput } from './XNumberInput';
export { default as XSelect } from './XSelect';
