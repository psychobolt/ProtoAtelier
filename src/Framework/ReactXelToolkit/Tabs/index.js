export { default as XTab } from './XTab.component';
