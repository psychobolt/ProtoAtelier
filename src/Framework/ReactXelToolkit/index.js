export * from './Forms';
export * from './Tabs';
