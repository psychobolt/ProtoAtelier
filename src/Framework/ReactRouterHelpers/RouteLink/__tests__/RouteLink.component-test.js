import React from 'react';
import { shallow } from 'enzyme';

import renderTest from 'Framework/ReactRouterRenderTest';
import { XTab } from 'Framework/ReactXelToolkit';
import RouteLink from '../RouteLink.component';

const ROOT = '/';

describe('component <RouteLink />', () => {
  it('should render correctly for no path', () => {
    const props = {
      to: [],
    };
    const wrapper = shallow(<RouteLink {...props}><XTab /></RouteLink>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly for a single path', () => {
    const props = {
      to: '/path',
    };
    const wrapper = shallow(<RouteLink {...props}><XTab /></RouteLink>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly for multiple paths with a default', () => {
    const props = {
      to: [{
        to: '/',
      }, {
        to: '/path',
        default: true,
      }],
    };
    const wrapper = shallow(<RouteLink {...props}><XTab /></RouteLink>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly for multiple paths with no default', () => {
    const props = {
      to: [{
        to: '/',
      }, {
        to: '/path',
      }],
    };
    const wrapper = shallow(<RouteLink {...props}><XTab /></RouteLink>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should go to path on click', () => {
    const props = {
      to: '/path',
    };
    const wrapper = renderTest(<RouteLink {...props}><XTab /></RouteLink>, {
      initialEntires: [ROOT],
      initialIndex: 0,
      steps: [
        ({ location }) => expect(location.pathname).toBe(ROOT),
        ({ location }) => expect(location.pathname).toBe(props.to),
      ],
    });
    wrapper.find(XTab).simulate('click');
  });
});
