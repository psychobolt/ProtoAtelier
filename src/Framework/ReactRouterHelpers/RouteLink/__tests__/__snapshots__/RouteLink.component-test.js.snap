// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <RouteLink /> should render correctly for a single path 1`] = `
<Route
  path="/path"
/>
`;

exports[`component <RouteLink /> should render correctly for multiple paths with a default 1`] = `
<Route
  path="(/|/path)"
/>
`;

exports[`component <RouteLink /> should render correctly for multiple paths with no default 1`] = `
<Route
  path="(/|/path)"
/>
`;

exports[`component <RouteLink /> should render correctly for no path 1`] = `
<Route
  path=""
/>
`;
